internal/cluster/pd_test.go

```
@@ -0,0 +1,213 @@
+	"sync"
+	"testing"
+	"time"
+
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// freePort grabs an ephemeral port and releases it for the next listener.
+// The bind-close-reuse window is small enough for in-process tests.
+func freePort(t *testing.T) int {
+	t.Helper()
+	l, err := net.Listen("tcp", "127.0.0.1:0")
```

golangci-lint (2.12.2)

[error] net.Listen must not be called. use (*net.ListenConfig).Listen

(noctx)